1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,91 @@
-function setAlarm() {}
+// Access to input
+const alermSetInput = document.querySelector("#alarmSet");
+// Access to span for showing timer
+const timeRemaining = document.querySelector(".timeRange");
+
+let intervalID;
+let intervalColor;
+
+function setAlarm() {
+  if (alermSetInput.value === "") {
+    alert("Please enter the proper input");
+  } else {
+    // Turn string value to number
+    let totalSeconds = alermSetInput.value;
+    // convert sec to 00:00 form
+    let timer = convertSecondsToMinutesAndSeconds(totalSeconds);
+    // Show timer in time remaining
+    timeRemaining.textContent = timer;
+    document.body.style.backgroundColor = "green";
+
+    // Set interval for timer
+    intervalID = setInterval(() => {
+      // Decrease time by 1 sec
+      timer = decrementByOneSec(timer);
+      if (timer === "00:00") {
+        // timeRemaining.textContent = timer;
+        document.body.style.backgroundColor = "red";
+        // Deactive interval
+        clearInterval(intervalID);
+        // Active alarm
+        playAlarm();
+        // stop alarm after 100 s
+        setTimeout(() => {
+          stopAlerm();
+        }, 100000);
+        intervalColor = setInterval(() => {
+          if (document.body.style.backgroundColor === "red") {
+            document.body.style.backgroundColor = "white";
+          } else {
+            document.body.style.backgroundColor = "red";
+          }
+        }, 500);
+      }
+    }, 1000);
+  }
+}
+
+// Convert number to a form of 00:00
+let convertSecondsToMinutesAndSeconds = function (totalSeconds) {
+  let minutes = Math.floor(totalSeconds / 60);
+  let seconds = totalSeconds % 60;
+  return (
+    (minutes < 10 ? "0" : "") +
+    minutes +
+    ":" +
+    (seconds < 10 ? "0" : "") +
+    seconds
```

Comment:
for reference, there's `padStart` method you can use for time. 

e.g
```
minutes.padStart(2, '0');
seconds.padStart(2, '0');

return `${minutes}:${seconds}`;
```

Reply:
:pray: